### ISSUE TITLE
@Property not showing in editor

### ISSUE THREAD

**Issue by al1-ce** [godot-dlang/godot-dlang#41]

**Comment by Superbelko** [comment]
Not sure what do you mean but possibly will be fixed with beta5 update.

Possible reason is that in register.d:505 there is a line like that, it has a comma instead of semicolon and that has effect of moving a type declaration only if that static if is true.

Very hard to track bug.

__register.d:505__:
```d
static if (Variant.variantTypeOf!P == VariantType.object)
    pinfo.class_name = godotName!(P).ptr,
pinfo.type = vt;
```

**Comment by Superbelko** [comment]
Ok even though in editor help now all properties are visible some of them are not listed in inspector, in test project `dreal` and `freal` are missing, probably a `number` too.

Just to be clear, it seems only auto properties are affected, i.e. normal D field with `@Property` attribute.

**Comment by al1-ce** [comment]
Godot properties are managed by godot itself
Need to look at cpp implementation to figure out what's going on

**Comment by Superbelko** [comment]
Ugh, I mean they are indeed got registered, but they are not showing in the inspector because of incorrect flags combination or something like that.

**Comment by al1-ce** [comment]
Oh, sorry, misunderstood
Still better check cpp to see what we have wrong

**Comment by SpiceyWolf** [comment]
As a fair warning, I made stuff in C++ where sometimes they arent working in the editor, I would have to restart the editor once or twice and sometimes with working code STILL not working, have to nuke the .godot folder and restart before they would work. Might check that process and confirm its actually definitely not working and not a godot bug before stressing too hard.

**Comment by Superbelko** [comment]
Property variables (`freal` and `dreal` in test project) should have correct flags set in `api.register.register()`

__api/register.d__:
```d
// somewhere under this foreach
static foreach (pName; godotPropertyVariableNames!T) {
    // ...
    import godot.globalenums : PropertyUsageFlags;
    pinfo.usage = PropertyUsageFlags.propertyUsageDefault; // <-- this!
    // ...
}
```

Last time I checked godot-cpp with debugger it has usage=7 which is `propertyUsageDefault | 1` but there is no flag for 1, anyway default is enough.

**Comment by Superbelko** [comment]
Fixed with #95
